src/dynamics/video_player/player/player.js

```diff
@@ -2677,22 +2676,46 @@ Scoped.define("module:VideoPlayer.Dynamics.Player", [
                                     return promise.asyncError(error);
                                 }, this) :
                                 console.log("Please define requestForTheNextAdTagURL method with Promise.");
-                        }, this, immediate ? 100 : requestInterval || 30000);
+                        }, this, immediate ? 100 : this.get("outstreamoptions.recurrenceperiod") || 30000);
                     }
                     return promise;
                 },
 
-                resetOutstreamRetries() {
-                    this.set("availableOutstreamRetries", this.get("outstreamoptions.numOfRetriesOnError") || 0);
+                setImmediateOutstreamRequests(value) {
+                    this.set("immediateOutstreamRequests", value);
+
+                    if (value) {
+                        this.set("availableOutstreamRetries", this.get("outstreamoptions.numOfRetriesOnError") || 0);
+                    } else {
+                        // Manually trigger a non-immediate ad request when we toggle to false.
+                        this.trigger("outstreamRetryOnInterval");
+                    }
+                },
+
+                handleOutstreamAdImpression() {
+                    const immediateRequests = this.get("immediateOutstreamRequests");
+                    // Reset `availableOutstreamRetries` for immediate outstream requests.
```

Comment:
Comment: i'd update this comment to add a bit more context.  It took me a couple of reads to realize you're resetting the available retries.  The function `setImmediateOutstreamRequests()` is misleading and should be something like `sendImmediateOutstreamAdRequest`